crates/iceberg/src/spec/snapshot.rs

```diff
@@ -124,6 +150,70 @@ impl Snapshot {
         Utc.timestamp_millis_opt(self.timestamp_ms).unwrap()
     }
 
+    /// Get the schema id of this snapshot.
+    #[inline]
+    pub fn schema_id(&self) -> Option<SchemaId> {
+        self.schema_id
+    }
+
+    /// Get the schema of this snapshot.
+    pub fn schema(&self, table_metadata: &TableMetadata) -> Result<SchemaRef> {
+        Ok(match self.schema_id() {
+            Some(schema_id) => table_metadata
+                .schema_by_id(schema_id)
+                .ok_or_else(|| {
+                    Error::new(
+                        ErrorKind::DataInvalid,
+                        format!("Schema with id {} not found", schema_id),
+                    )
+                })?
+                .clone(),
+            None => table_metadata.current_schema().clone(),
+        })
+    }
```

Comment:
This method is used to fetch the schema of a snapshot, so I think this is compatible with java/python behavior?

Reply:
In Python we just fixed it here: https://github.com/apache/iceberg-python/pull/247

The thing is that the `current-schema` can be pointing to a newer snapshot than the snapshot-id. This is because schema and data are tracked independently in Iceberg. However, if you just want to see the latest version of the table, you want to use the `current-snapshot`. When you want to see the table at a certain snapshot, then you want to use the schema-id from the snapshot. 

I think we can also do this in a separate PR.